Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -924,23 +934,28 @@ extension TaskHandler: ChannelDuplexHandler {
         let response = self.unwrapInboundIn(data)
         switch response {
         case .head(let head):
-            if !head.isKeepAlive {
-                self.closing = true
-            }
+            switch self.state {
```

Comment:
This may read more clearly as:

```swift
if case .endOrError = self.state {
    return
}
```

Rather than forcing the indentation of the rest of the body.

Reply:
ah! thank you, always forget about this pattern

Reply:
Seems like the only place you didn't apply this pattern was this block? :wink:

Reply:
😅